storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteIndexCache.java

```diff
@@ -208,46 +203,32 @@ public void removeAll(Collection<Uuid> keys) {
         }
     }
 
-    private void enqueueEntryForCleanup(Entry entry, Uuid key) {
+    private void enqueueEntryForCleanup(Entry entry) {
         try {
             entry.markForCleanup();
-            if (!expiredIndexes.offer(entry)) {
-                log.error("Error while inserting entry {} for key {} into the cleaner queue because queue is full.", entry, key);
-            }
+            Runnable runnable = () -> {
+                try {
+                    log.debug("Cleaning up index entry {}", entry);
+                    entry.cleanup();
+                } catch (Exception ex) {
+                    // do not exit for exceptions other than InterruptedException
+                    log.error("Error occurred while cleaning up expired entry", ex);
+                }
+            };
+            cleanerScheduler.scheduleOnce("delete-index", runnable, fileDeleteDelayMs);
```

Comment:
Good catch on the earlier issue and fixing it using a schedule delay like we do with local segments.